### PR TITLE
Don't parse .content if the template has none. Fixes #166

### LINKED ISSUE #166: Error scoping <template> in <svg> context
When a `<template>` element is parsed in `<svg>` context, it is not actually an `HTMLTemplateElement` with a `.content` property, and this code called from the context of `prepareTemplate` on such a template throws trying to dereference it: https://github.com/webcomponents/shadycss/blob/master/src/style-transformer.js#L59

Since downstream code may want to "prollyfill" a `<template>` in svg context to turn it into an actual template with content, `shadycss` should be defensive to `<template>` elements that are not `HTMLTemplateElement`s.

### Repro:
http://jsbin.com/himoceb/edit?html,console,output

### PR DESCRIPTION
**Pull request by kevinpschaaf**
Fixes #166 

Added tests for nested template parsing (both HTML and SVG), since that seemed to be untested.

### CODE REVIEW

**Comment by TimvdLippe** [comment]
Firefox is passing. Chrome tests are failing due to selenium. It seems that the tests do not work in Safari, as Safari has the display inline instead of inline-block.

Code-wise LGTM

File: src/style-transformer.js [hunk 1]
```diff
@@ -56,7 +56,9 @@ class StyleTransformer {
       this.element(node, selector, shouldRemoveScope);
     }
     let c$ = (node.localName === 'template') ?
-      (node.content || node._content).childNodes :
+      // In case the template is in svg context, fall back to the node
+      // since it won't be an HTMLTemplateElement with a .content proeprty
```
Comment (valdrinkoshi) [inline src/style-transformer.js]: nit: s/proeprty/property

**Comment by azakus** [comment]
@kevinpschaaf looks like there are real failures on Edge, Safari 10, and Chrome 41

**Comment by kevinpschaaf** [comment]
The Edge failure is due to the fact that by introducing `<template>` in SVG, the Template polyfill now needs to differentiate between fixing templates with busted contents (due to Edge template bugs) vs templates in SVG which will not have content (because they are not `HTMLTemplateElement`).

**Comment by kevinpschaaf** [comment]
Per above, this is blocked on https://github.com/webcomponents/template/pull/35 and should be re-run after that is merged.

**Comment by TimvdLippe** [comment]
The template fix has been merged.

**Comment by TimvdLippe** [comment]
The Edge tests have been fixed, but now IE11 and Chrome 41 started failing :cry:

**Comment by TimvdLippe** [comment]
@kevinpschaaf I have fixed the tests. 2 issues:

1. In IE11, `element.id` for a `template` inside a `svg` returns `undefined`. We had to use `element.getAttribute('id')` instead
1. In both IE11 and Chrome 11, the second parameter of `Array.from` is ignored. Instead perform the `Array.from` first and then call `.map` manually

**Pull request merged**